cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -107,30 +132,80 @@ function mapDispatchToProps(dispatch: any): DispatchToProps {
 
 type Props = StateToProps & DispatchToProps;
 class AnnotationTopBarContainer extends React.PureComponent<Props> {
+    private autoSaveInterval: number | undefined;
+
     public componentDidUpdate(): void {
         const {
             jobInstance,
             frameNumber,
+            frameChangeTime,
+            frameSpeed,
             playing,
             canvasIsReady,
-            onChangeFrame,
             onSwitchPlay,
         } = this.props;
 
+        
         if (playing && canvasIsReady) {
             if (frameNumber < jobInstance.stopFrame) {
+                const delay: number = frameChangeTime ? Math.max(0,~~(1000/frameSpeed) -  new Date().getTime() + frameChangeTime) : 0;
```

Comment:
Let's use Math.floor because it acts more obvious than double NOT bit wise operator